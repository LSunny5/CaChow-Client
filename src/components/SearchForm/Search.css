/*search page*/
.SearchContent {
    background: url(/images/editedrestaurant.jpg) no-repeat center center fixed; 
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
    min-height: 92vh;
    padding-bottom: 1rem;
}

/*heading title*/
.searchTitle {
    font-family: 'Cinzel', serif;
    font-size: 2em;
    margin: 1rem;
}

/*search text field*/
#searchBar {
    width: 70%;
    font-size: 1.5em;
    margin: 15px auto 0;
    border: none;
    border-bottom: 2px solid rgb(16, 73, 16);
    border-radius: 3px;
    font-family: 'Delius', cursive;
    padding: 5px;
}

/*box for search results*/
.resultList {
    list-style: none;
    padding: 0;
    display: flex;
    flex-direction: column;    
}

/*restaurant button front*/
.resultButton {
    width: 60%;
    margin: 7px auto;
    font-family: 'Delius', cursive;
    text-decoration: none;
    line-height: 25px;
    letter-spacing: 1px;
    position: relative;
}

/*after view of button*/
.resultButton:hover:after {
    opacity: 1;
    transform: translateY(0) rotateX(0);
    
}

/*front of button rotating down*/
.resultButton:hover:before {
    transform: translateY(50%) rotateX(90deg);
}

/*button before*/
.resultButton:before {
    top: 0;
    left: 0;
    opacity: 1;
    color: white;
    transition: 0.5s;
    background: #323237;
    content: attr(data-front);
    line-height: 5rem;
    display: block;
    position: relative;
    transform: translateY(0) rotateX(0);
    padding: 0 30px;
    border: 1px solid black;
}

/*button look on hover*/
.resultButton:after {
    top: 0;
    left: 0;
    position: absolute;
    opacity: 0;
    width: 100%;
    transition: 0.5s;
    transform: translateY(-50%) rotateX(90deg);
    background: url(/images/spread.jpeg);
    background-size: cover;
    content: attr(data-back);
    line-height: 5rem;
    border: 1px solid black;
    color: white;
    font-weight: bolder;
}

/********************************************************************/
/*For Ipads and larger screens */
@media only screen and (min-width: 600px) {
    /*search text field*/
    #searchBar {
        width: 40%;
    }

    /*box for search results*/
    .resultList {
        flex-direction: row;
        flex-wrap: wrap;
        margin: 5px auto;
        width: 80%;
    }

    /*restaurant button front*/
    .resultButton {
        display: block;
        margin: 10px auto;
        width: 40%;
    }

    /*button before*/  /*button look on hover*/
    .resultButton:before, .resultButton:after {
        line-height: 6rem;
    }
}

/********************************************************************/
/*For Laptops and larger screens */
@media only screen and (min-width: 1024px) {
    /*search text field*/
    #searchBar {
        width: 30%;
    }

    /*box for search results*/
    .resultList {
        flex-direction: row;
        flex-wrap: wrap;
        margin: 5px auto;
        width: 80%;
    }

    /*restaurant button front*/
    .resultButton {
        display: block;
        margin: 10px auto;
        width: 30%;
    }

    /*button before*/  /*button look on hover*/
    .resultButton:before, .resultButton:after {
        line-height: 6rem;
    }
}