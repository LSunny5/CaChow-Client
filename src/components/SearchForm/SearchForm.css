.SearchContent {
    width: 80%;
    margin: 0 auto;
}

#searchBar {
    width: 75%;
    font-size: 1.3em;
    margin: 10px auto;
    border: none;
    border-bottom: 1px solid blue;
}

.searchButton {
    /* text-decoration: none;
    color: black;
    border: 1px solid black;
    padding: 0 30px;
    margin: 0 10px;
    font-size: 1.2em;
    cursor: pointer; */
    
    width: 2.2rem;
    height: 2.2rem;
}

.searchImage {
    width: 1.5rem;
    height: 1.5rem;
    margin: 0 10px;
}