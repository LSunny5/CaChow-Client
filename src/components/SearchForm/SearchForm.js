import React from 'react';
import './Search.css';
import PropTypes from 'prop-types';

class SearchForm extends React.Component {
    updateSearchTerm = (event) => {
        this.props.updateSearch(event.target.value);
    }

    render() {
        return (
            <form>
                <label htmlFor="searchBar" className='labelDesign'>
                    <input type="text"
                        id="searchBar"
                        placeholder="Search..."
                        name="searchBar"
                        className="searchText"
                        autoComplete="off"
                        onChange={this.updateSearchTerm} >
                    </input>
                </label>
            </form>
        );
    };
}

SearchForm.propTypes = {
    updateSearch:  PropTypes.func, 
}

export default SearchForm;