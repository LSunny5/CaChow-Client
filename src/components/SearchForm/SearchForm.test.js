

import React from 'react';
import ReactDOM from 'react-dom';
import SearchForm from './SearchForm';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`SearchForm component`, () => {
    const props = {
        updateSearch: () => { },
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <SearchForm {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the SearchForm form', () => {
        const wrapper = shallow(<SearchForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});