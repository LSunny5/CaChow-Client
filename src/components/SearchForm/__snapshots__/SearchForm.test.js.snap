// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders the SearchForm form 1`] = `
<form>
  <label
    className="labelDesign"
    htmlFor="searchBar"
  >
    <input
      autoComplete="off"
      className="searchText"
      id="searchBar"
      name="searchBar"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
    />
  </label>
</form>
`;
